### PR TITLE
refactor(experimental): errors: `compat` package

### PR DESCRIPTION
**Pull request by buffalojoec**
Adds custom `SolanaError` throws to the `@solana/compat` package.

### CODE REVIEW

**Comment by buffalojoec** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2202?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#2202** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2202?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2192** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2192?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2191** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2191?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2190** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2190?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2189** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2189?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2201** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2201?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2200** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2200?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2188** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2188?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2187** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2187?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2186** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2186?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2185** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2185?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2184** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2184?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2182** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2182?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#2183](https://github.com/solana-labs/solana-web3.js/pull/2183) <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2183?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `master`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2202?utm_source=stack-comment).

File: packages/compat/src/transaction.ts [hunk 1]
```diff
@@ -99,15 +108,16 @@ export function fromVersionedTransactionWithBlockhash(
     // - will need to convert account instructions to `IAccountLookupMeta` when appropriate
     if (transaction.message.addressTableLookups.length > 0) {
         // TODO coded error
+        // This should probably not be a `SolanaError`, since we need to add
+        // this functionality.
```
Comment (steveluscher) [inline packages/compat/src/transaction.ts:112]: 👍🏻

File: packages/errors/src/messages.ts [hunk 2]
```diff
@@ -137,6 +141,12 @@ export const SolanaErrorMessages: Readonly<{
         'here: https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts',
     [SOLANA_ERROR__SUBTLE_CRYPTO_SIGN_FUNCTION_MISSING]: 'No signing implementation could be found',
     [SOLANA_ERROR__SUBTLE_CRYPTO_VERIFY_FUNCTION_MISSING]: 'No key export implementation could be found',
+    [SOLANA_ERROR__TRANSACTION_INVALID_NONCE_TRANSACTION_FIRST_INSTRUCTION_NOT_ADVANCE_NONCE]:
+        'Transaction first instruction is not advance nonce account instruction.',
```
Comment (steveluscher) [inline packages/errors/src/messages.ts:145]: I know I almost certainly put these periods here, but we should probably either _always_ end error messages with periods, or _never_ end them with periods.

**Review by steveluscher — APPROVED** [review]
Added via Giphy

**Comment by steveluscher** [comment]
### Merge activity

* **Feb 29, 11:33 AM PST**: @steveluscher started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2202).

**Pull request merged**